src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    public class JsonContent : HttpContent
+    {
+        internal const string JsonMediaType = "application/json";
+
+        private readonly byte[] _content;
+        private readonly int _offset;
+        private readonly int _count;
+
+        private static MediaTypeHeaderValue CreateMediaType(string mediaTypeAsString)
+        {
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(mediaTypeAsString);
```

Comment:
The Formatting API uses the constructor to create the media type from the string argument.
As a consequence, the following fails because the charset is not accepted:
```cs
new JsonContent(typeof(Foo), foo, mediaType: "foo/bar; charset=utf-16");
```

Alternatively we use `MediaTypeHeaderValue.Parse` to make previous call work out-of-box.

Reply:
`StringContent` exposes this via overloads:

```c#
public StringContent (string content, MediaTypeHeaderValue mediaType);
public StringContent (string content, System.Text.Encoding encoding);
public StringContent (string content, System.Text.Encoding encoding, string mediaType);
```

We should be consistent here.

Reply:
All right, `StringContent` also uses the `MediaTypeHeaderValue` constructor. It seems fair enough to align with other derived types.

I am taking the suggestion, thanks.